### DISCUSSION TITLE
Panel

### DISCUSSION THREAD

**Discussion by bameyrick** [ONSdigital/design-system#49] [Feedback]
Use this place to discuss the [panel component](https://ons-design-system.netlify.com/components/panel) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by bameyrick** [comment]
Links within panels should not have their colour modified. Links should always be the same blue everywhere.

**Comment by jrbarnes9** [comment]
Panels need a default bottom margin parameter

**Comment by bameyrick** [comment]
Maybe we should address all all of these issues when we look at spacing: https://trello.com/c/h0RbO4UV/99-spacing

**Comment by bameyrick** [comment]
Theres no classes parameter so adding custom spacing via utility classes isn't possible

**Comment by bameyrick** [comment]
We need a parameter to customise the tag used within the header element, e.g. to use a `h1` for the title.

**Comment by jrbarnes9** [comment]
Reintroduce a 'warning' panel when the consequences of the users action could have a negative effect.  e.g. The EQ 'remove person' confirmation message: 

<img width="795" alt="Screenshot 2019-11-08 at 15 49 13" src="https://user-images.githubusercontent.com/43346934/68489605-5aee8e00-023f-11ea-8969-e1bdd58d21ea.png">

**Comment by jrbarnes9** [comment]
Remove `border-left` from warning panel

**Comment by trevorsaint** [comment]
> Remove `border-left` from warning panel

Yeah I’ve spotted this bug and need it removed.

![image](https://user-images.githubusercontent.com/4989027/86143792-cba72b00-baec-11ea-9eb5-174a56ad07b9.png)

**Comment by jrbarnes9** [comment]
Warning panel text should be bold by default
![Screenshot 2020-07-09 at 16 06 29](https://user-images.githubusercontent.com/43346934/87056917-2c75e800-c1fe-11ea-8c2b-7548bcaa338c.png)

**Comment by jrbarnes9** [comment]
Wrapping issue with error summary panel

![Screenshot 2020-07-30 at 10 08 23](https://user-images.githubusercontent.com/43346934/88904247-bf79cf00-d24c-11ea-94d0-bc40b3f8e9f9.png)

**Comment by jrbarnes9** [comment]
Need to be able to define a panel icon size

<img width="721" alt="Screenshot 2020-08-05 at 15 51 45" src="https://user-images.githubusercontent.com/43346934/89427971-c2d0f700-d733-11ea-8bbe-96ce17922f7d.png">

**Comment by jrbarnes9** [comment]
Wrapping issue in error summary panel...
 
![image](https://user-images.githubusercontent.com/43346934/93993532-c4f42100-fd86-11ea-9132-720f3c37d168.png)

**Comment by jrbarnes9** [comment]
Remove `border-left` from `panel--bare`

`panel--bare` should have no left border, so e.g. icons in these panels can be left-aligned with the page container.
![Screenshot 2020-10-09 at 16 39 35](https://user-images.githubusercontent.com/43346934/95603345-0b67a200-0a4e-11eb-9d53-e8d30aa21933.png)

**Comment by jrbarnes9** [comment]
Remove top and bottom padding from `panel--bare` so spacing is dictated by bottom margins

**Comment by jrbarnes9** [comment]
Adjust panel icon vertical alignment

![Screenshot 2020-10-29 at 09 35 09](https://user-images.githubusercontent.com/43346934/97550993-4ebc8d00-19ca-11eb-95e7-525188f765aa.png)

**Comment by jrbarnes9** [comment]
Remove unnecessary padding from `panel--warn`

<img width="975" alt="Screenshot 2020-11-27 at 08 24 55" src="https://user-images.githubusercontent.com/43346934/100427492-70bd3400-308a-11eb-8791-091756e6f518.png">

**Comment by trevorsaint** [comment]
Spotted a little bug in the Information Panel example. 

We are wrapping the UL inside a P tag.

<img width="525" alt="Screenshot 2021-01-12 at 08 10 14" src="https://user-images.githubusercontent.com/4989027/104286991-b5bbef00-54ad-11eb-83aa-91029b0dd68b.png">

**Comment by trevorsaint** [comment]
The panel announcement will need to change to be more inline with GOV.

Large

<img width="1068" alt="large" src="https://user-images.githubusercontent.com/4989027/114191488-60746a00-9944-11eb-92c5-c3a7b7856d36.png">

Small

<img width="1034" alt="small" src="https://user-images.githubusercontent.com/4989027/114191510-68cca500-9944-11eb-8c65-6c7f58f470a2.png">

**Comment by rmccar** [comment]
Do we want all success panels to have the tick icon hardcoded? Because at the moment it is possible to not specify an icon for a success panel like this:

![image](https://user-images.githubusercontent.com/42928680/160568031-1c34d423-d7c9-45b0-9e1b-261cb7c741b3.png)

It is also possible to add an icon to the information panel for example and it is not coloured specifically for that panel, like this:

![image](https://user-images.githubusercontent.com/42928680/160568301-b989a151-c423-4b5d-97f4-766603977680.png)

**Comment by jrbarnes9** [comment]
DAC: Add params to add attributes to a success panel to ensure screen readers can read it first. (Role, aria-labelledby)

**Reply by jrbarnes9** [reply]
#2340

**Comment by LaurenDowling** [comment]
There doesn't appear to be a success/confirmation page within the patterns section of the design system yet. I was just wondering whether it is advised that this component sit on a success/confirmation page? The example below shows a case management system mockup where 'confirm and send email' has been clicked on a previous page, ending that particular journey and returning the user to the respondent case page. So it behaves more like a notification banner.
![image](https://user-images.githubusercontent.com/108019533/194107317-175cda6e-1749-45de-9c63-e8d39e78a703.png)

If this use is fine, has the potential to close the banner been considered? 

I noticed that the Scottish Government design system seem to have both a notification banner (that can be closed) and a notification panel. https://designsystem.gov.scot/components/notification-banner/

**Reply by jrbarnes9** [reply]
Hi @LaurenDowling. This is on our radar. We do have a ticket on our community backlog to publish the [confirmation page pattern](https://github.com/ONSdigital/design-system/issues/1643) that was used ending linear transactional journeys, however, we do need more research undertaken/shared for applications like this, where there is repetitive use, to understand if the confirmation page would work in these cases. I would imagine the journey to return via a call to action may not work so well here. 

If anyone has any research they could share, please do. If you feel a close button is required, please test it and propose a change.

We also plan to do a design spike on notifications ourselves in the future, as we are aware they are becoming more widely used within services using the DS now, but we encourage all designers to contribute to the DS where possible.

**Comment by Al-Davies** [comment]
**Panel warning - UR findings**
tested most recently as part of end-to-end testing Feb-Dec 2021 of Census

- Users have clearly noted panels used for confirmation messages as part of 2021 Census testing
- Users have clearly noted blue warning panels used on 2021 Census home page (displaying information on fines for non-completion), blue banners described as 'striking'
- Users notice panels and warnings, but content must to be the point and relevant, if they aren't they 'get in the way'
